cmd/dagger/config.go

```diff
@@ -0,0 +1,607 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"dagger.io/dagger"
+	"github.com/dagger/dagger/core/modules"
+	"github.com/dagger/dagger/dagql/idtui"
+	"github.com/dagger/dagger/engine/client"
+	"github.com/juju/ansiterm/tabwriter"
+	"github.com/muesli/termenv"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"github.com/vito/progrock"
+)
+
+var configJSONOutput bool
+
+func init() {
+	configCmd.PersistentFlags().BoolVar(&configJSONOutput, "json", false, "output in JSON format")
+	configCmd.AddGroup(moduleGroup)
+
+	configCmd.AddCommand(
+		configIncludeCmd,
+		configViewsCmd,
+	)
+}
+
+var configCmd = &cobra.Command{
+	Use:   "config",
+	Short: "Get or set the configuration of a Dagger module",
+	Long:  "Get or set the configuration of a Dagger module. By default, print the configuration of the specified module.",
+	Example: strings.TrimSpace(`
+dagger config -m /path/to/some/dir
+dagger config -m github.com/dagger/hello-dagger
+`,
+	),
+	Args:    cobra.NoArgs,
+	GroupID: moduleGroup.ID,
+	RunE: configSubcmdRun(func(ctx context.Context, cmd *cobra.Command, _ []string, modConf *configuredModule) (err error) {
+		ctx, vtx := progrock.Span(ctx, idtui.PrimaryVertex, cmd.CommandPath())
+		defer func() { vtx.Done(err) }()
+		cmd.SetContext(ctx)
+		setCmdOutput(cmd, vtx)
+
+		if configJSONOutput {
+			cfgContents, err := modConf.Source.Directory(".").File(modules.Filename).Contents(ctx)
+			if err != nil {
+				return fmt.Errorf("failed to read module config: %w", err)
+			}
+			cmd.OutOrStdout().Write([]byte(cfgContents))
+			return nil
+		}
+
+		mod := modConf.Source.AsModule()
+
+		name, err := mod.Name(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to get module name: %w", err)
+		}
+		sdk, err := mod.SDK(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to get module SDK: %w", err)
+		}
+
+		tw := tabwriter.NewWriter(cmd.OutOrStdout(), 0, 0, 3, ' ', tabwriter.DiscardEmptyColumns)
+		fmt.Fprintf(tw, "%s\t%s\n",
+			"Name:",
+			name,
+		)
+		fmt.Fprintf(tw, "%s\t%s\n",
+			"SDK:",
+			sdk,
+		)
+		fmt.Fprintf(tw, "%s\t%s\n",
+			"Root Directory:",
+			modConf.LocalContextPath,
+		)
+		fmt.Fprintf(tw, "%s\t%s\n",
+			"Source Directory:",
+			modConf.LocalRootSourcePath,
+		)
+
+		return tw.Flush()
+	}).RunE,
+}
+
+var configIncludeCmd = configSubcmd{
+	Use:   "include",
+	Short: "Get or set the include paths of a Dagger module",
+	Long:  "Get or set the include paths of a Dagger module. By default, print the include paths of the specified module.",
+
+	Hidden: true,
```

Comment:
Add the experimental annotation so it's not included when generating the cli reference.

Reply:
It looks like the cli reference didn't include the hidden subcommands already actually, but added the experimental annotation anyways in case I'm missing something

Reply:
Ah, right. Some of the hidden ones we wanted on the reference, but not others. So we removed the hidden flag and added the experimental in those. In this case, it should be just hidden, sorry about that.